contracts/src/utils/structs/bitmap.rs

```diff
@@ -0,0 +1,140 @@
+//! Contract module for managing `U256` to boolean mapping in a compact and
+//! efficient way, provided the keys are sequential. Largely inspired by Uniswap's <https://github.com/Uniswap/merkle-distributor/blob/master/contracts/MerkleDistributor.sol>[merkle-distributor].
+//!
+//! `BitMap` packs 256 booleans across each bit of a single 256-bit slot of
+//! `U256` type. Hence, booleans corresponding to 256 _sequential_ indices
+//! would only consume a single slot, unlike the regular boolean which would
+//! consume an entire slot for a single value.
+//!
+//! This results in gas savings in two ways:
+//!
+//! - Setting a zero value to non-zero only once every 256 times
+//! - Accessing the same warm slot for every 256 _sequential_ indices
+use alloy_primitives::U256;
+use stylus_proc::sol_storage;
+
+sol_storage! {
+    /// State of bit map.
+    pub struct BitMap {
+        /// Inner laying mapping.
+        mapping(uint256 => uint256) _data;
+    }
+}
+
+impl BitMap {
+    /// Returns whether the bit at `index` is set.
+    ///
+    /// # Arguments
+    ///
+    /// * `index` - index of boolean value at the bit map.
+    #[must_use]
+    pub fn get(&self, index: U256) -> bool {
+        let bucket = index >> 8;
+        let mask = U256::from(1) << (index & U256::from(0xff));
```

Comment:
imho you can reuse here `self.get_mask(index)`

Reply:
thnx!